scripts/powershell/version-repo-with-tag.ps1

```diff
@@ -0,0 +1,139 @@
+<#
+.DESCRIPTION
+Tags latest or a specific commit in the specified Repo.
+Uses a combination of the Repo Name, Current Date and a Version number to create the tag.
+Tag serve as a means of versioning the entire repo.
+.PARAMETER RepoName
+The name of the repository. EG "azure-sdk-tools"
+.PARAMETER RepoOrg
+The owning organization of the repository. EG "Azure"
+.PARAMETER RepoType
+Should be either a github or git (Azure) repo
+.PARAMETER CommitSHA
+The commitSHA to tag. Defaults to master
+.PARAMETER RepoProjectID
+For git Repos supply the project ID. Default to the internal project ID set by the pipeline
+.PARAMETER PATForAPI
+Personal Access Token for authentication to the Repo
+#>
+param (
+    [Parameter(Mandatory = $true)]
+    [string] $RepoName,
+
+    [Parameter(Mandatory = $true)]
+    [string] $RepoOrg,
+
+    [Parameter(Mandatory = $true)]
+    [ValidateSet('github', 'git')]
+    [string] $RepoType,
+
+    [string] $CommitSHA="master",
+
+    [Parameter(Mandatory = $true)]
+    [string] $RepoProjectID,
+
+    [Parameter(Mandatory = $true)]
+    [string] $GitHubPAT,
+    [Parameter(Mandatory = $true)]
+    [string] $AzureGitPAT
+)
+
+Write-Host $MyInvocation.Line
+
+$HttpHeaders = @{}
+
+if ($RepoType -eq 'github')
```

Comment:
I think the logic for creating the tag can go into your shared powershell module this script should simply be about creating the tag format and calling those helpers to create the tag.